pilot/pkg/config/kube/crd/client.go

```diff
@@ -220,34 +273,57 @@ func (cl *Client) RegisterResources() error {
 
 // DeregisterResources removes third party resources
 func (cl *Client) DeregisterResources() error {
-	clientset, err := apiextensionsclient.NewForConfig(cl.restconfig)
+	for k, rc := range cl.clientset {
+		log.Infof("deregistering for apiVersion ", k)
+		if err := rc.deregisterResources(); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (rc *restClient) deregisterResources() error {
+	cs, err := apiextensionsclient.NewForConfig(rc.restconfig)
 	if err != nil {
 		return err
 	}
 
 	var errs error
-	for _, schema := range cl.descriptor {
+	for _, schema := range rc.descriptor {
 		name := ResourceName(schema.Plural) + "." + ResourceGroup(&schema)
-		err := clientset.ApiextensionsV1beta1().CustomResourceDefinitions().Delete(name, nil)
+		err := cs.ApiextensionsV1beta1().CustomResourceDefinitions().Delete(name, nil)
 		errs = multierror.Append(errs, err)
 	}
 	return errs
 }
 
 // ConfigDescriptor for the store
 func (cl *Client) ConfigDescriptor() model.ConfigDescriptor {
-	return cl.descriptor
+	d := make(model.ConfigDescriptor, 0, len(cl.clientset))
+	for _, rc := range cl.clientset {
+		d = append(d, rc.descriptor...)
+	}
+	return d
 }
 
 // Get implements store interface
 func (cl *Client) Get(typ, name, namespace string) (*model.Config, bool) {
-	schema, exists := cl.descriptor.GetByType(typ)
+	s, ok := knownTypes[typ]
+	if !ok {
```

Comment:
Shouldn't unknown types return an error?

Reply:
Is there a reason we can't return the error here, instead of logging and returning nothing?